src/bootstrap/dist.rs

```diff
@@ -87,6 +87,39 @@ impl Step for Docs {
     }
 }
 
+#[derive(Debug, PartialOrd, Ord, Copy, Clone, Hash, PartialEq, Eq)]
+pub struct JsonDocs {
+    pub host: TargetSelection,
+}
+
+impl Step for JsonDocs {
+    type Output = Option<GeneratedTarball>;
+    const DEFAULT: bool = true;
+
+    fn should_run(run: ShouldRun<'_>) -> ShouldRun<'_> {
+        let alias = "rust-json-docs";
+        let default = should_build_extended_tool(&run.builder, alias);
+        run.alias(alias).default_condition(default)
+    }
+
+    fn make_run(run: RunConfig<'_>) {
+        run.builder.ensure(JsonDocs { host: run.target });
+    }
+
+    /// Builds the `rust-json-docs` installer component.
+    fn run(self, builder: &Builder<'_>) -> Option<GeneratedTarball> {
```

Comment:
Oh, I forgot - please gate this on `builder.unstable_features()` - see https://github.com/rust-lang/rust/blob/491075c9c65652590ef34d2021136dfb747b3435/src/bootstrap/dist.rs#L1197-L1202 as an example.